File: java/simulation/field/grassField/RandomData.java
package simulation.field.grassField;

import simulation.common.PolarCoordinate;

public class RandomData {
	
	public float generateSpeed() {
		return (float) (Math.random()*3);	}

	public int generateAngle(){
		return (int) (Math.random()*(359));
		}
	
	public PolarCoordinate getRandomPolarCoordinate(){
		return new PolarCoordinate(generateSpeed(), generateAngle());
	}
	
	public PolarCoordinate[] getArray(){
		int size = (int)(Math.random() * 31);
		PolarCoordinate[] p = new PolarCoordinate[size];
		
		for(int i = 0; i < size; i++){
			p[i] = getRandomPolarCoordinate();
		}
		return p;
	}

}

File: src/simulation/field/grassField/RandomData.ts
import { PolarCoordinate } from '../../common/PolarCoordinate';

export class RandomData {
  generateSpeed(): number {
    return Math.random() * 3;
  }

  generateAngle(): number {
    return Math.floor(Math.random() * 360);
  }

  getRandomPolarCoordinate(): PolarCoordinate {
    return new PolarCoordinate(this.generateSpeed(), this.generateAngle());
  }

  getArray(): PolarCoordinate[] {
    const size = Math.floor(Math.random() * 31);
    const p: PolarCoordinate[] = [];
    for (let i = 0; i < size; i++) {
      p.push(this.getRandomPolarCoordinate());
    }
    return p;
  }
}
